odoo/addons/base/tests/test_expression.py

```diff
@@ -196,6 +196,40 @@ def test_10_hierarchy_in_m2m(self):
             cats = self._search(Category, [('id', 'parent_of', False)])
         self.assertEqual(len(cats), 0)
 
+    @mute_logger('odoo.models.unlink')
+    def test_10_hierarchy_access(self):
+        Partner = self.env['res.partner'].with_user(self.user_demo)
+        top = Partner.create({'name': 'Top'})
+        med = Partner.create({'name': 'Medium', 'parent_id': top.id})
+        bot = Partner.create({'name': 'Bottom', 'parent_id': med.id})
+
+        # restrict access of user Demo to contacts Big and Small
```

Comment:
big and small?

Reply:
Should be "top" and "bottom", thanks :wink: